src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -317,7 +323,14 @@ private Task RestartAsync()
         {
             if (!_shutdownScheduled && Interlocked.Exchange(ref _restartRequested, 1) == 0)
             {
-                return _scriptHostManager.RestartHostAsync();
+                if (Interlocked.Read(ref _suspensionRequestsCount) == 0)
+                {
+                    return _scriptHostManager.RestartHostAsync();
+                }
+                else
+                {
+                    _logger.LogDebug("Restart is triggered when notifications are suspended");
```

Comment:
nit: The message should be consistent with what we call the API, in the current case, if the API is called `SuspendRestart` the message should be something like: `Restart requested while currently suspended. Ignoring request.`